tracer-core/src/main/java/org/zalando/tracer/PhraseGenerator.java

```diff
@@ -0,0 +1,468 @@
+        "visvesvaraya",
+        // Christiane Nüsslein-Volhard - German biologist, won Nobel Prize in Physiology or Medicine in 1995 for research on the genetic control of embryonic development. https://en.wikipedia.org/wiki/Christiane_N%C3%BCsslein-Volhard
+        "volhard",
+        // Marlyn Wescoff - one of the original programmers of the ENIAC. https://en.wikipedia.org/wiki/ENIAC - https://en.wikipedia.org/wiki/Marlyn_Meltzer
+        "wescoff",
+        // Roberta Williams, did pioneering work in graphical adventure games for personal computers, particularly the King's Quest series. https://en.wikipedia.org/wiki/Roberta_Williams
+        "williams",
+        // Sophie Wilson designed the first Acorn Micro-Computer and the instruction set for ARM processors. https://en.wikipedia.org/wiki/Sophie_Wilson
+        "wilson",
+        // Jeannette Wing - co-developed the Liskov substitution principle. - https://en.wikipedia.org/wiki/Jeannette_Wing
+        "wing",
+        // Steve Wozniak invented the Apple I and Apple II. https://en.wikipedia.org/wiki/Steve_Wozniak
+        "wozniak",
+        // The Wright brothers, Orville and Wilbur - credited with inventing and building the world's first successful airplane and making the first controlled, powered and sustained heavier-than-air human flight - https://en.wikipedia.org/wiki/Wright_brothers
+        "wright",
+        // Rosalyn Sussman Yalow - Rosalyn Sussman Yalow was an American medical physicist, and a co-winner of the 1977 Nobel Prize in Physiology or Medicine for development of the radioimmunoassay technique. https://en.wikipedia.org/wiki/Rosalyn_Sussman_Yalow
+        "yalow",
+        // Ada Yonath - an Israeli crystallographer, the first woman from the Middle East to win a Nobel prize in the sciences. https://en.wikipedia.org/wiki/Ada_Yonath
+        "yonath"
+    };
+
+    // Carefully crafted list of neutral short verbs
+    private static final String[] VERBS = {
+        "asks",
+        "pays",
+        "sees",
+        "waits",
+        "calls",
+        "feeds",
+        "helps",
+        "knows",
+        "likes",
+        "meets",
+        "saves",
+        "tells"
+    };
+
+    private static final String[][] PARTS = {ADJECTIVES, SURNAMES, VERBS, ADJECTIVES, SURNAMES};
+
+    /*
+     * The random number generator used by this class to create random numbers.
+     * In a holder class to defer initialization until needed.
+     */
+    private static class Holder {
+        static final SecureRandom RANDOM = new SecureRandom();
+    }
+
+    /**
+     * @return a random name from the list of adjectives, surnames and verbs
+     * formatted as "adjective_surname_verb_adjective_surname".
+     * For example 'amazing_knuth_knows_romantic_liskov'.
+     */
+    @Override
+    public String generate() {
+        return generate(Holder.RANDOM::nextInt);
+    }
+
+    @VisibleForTesting
+    static String generate(final IntUnaryOperator r) {
+        do {
```

Comment:
If it's an infinite loop, `do-while` has no inherent benefit over `while`, has it?